### PR TITLE
[CodeWhisperer] Fix that a couple of document change user actions should cancel CodeWhisperer's popup.

### PR DESCRIPTION
**Pull request by andrewyuq**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Screenshots (if appropriate)
Before:

https://user-images.githubusercontent.com/89420755/181619502-99493b86-595d-48b7-bae3-add8a6cb788d.mov

After: 

https://user-images.githubusercontent.com/89420755/181621569-488cf2d2-6d51-4621-bd5f-477bfd119aa6.mov

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if change is customer facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt [hunk 1]
```diff
@@ -420,6 +439,16 @@ class CodeWhispererPopupManager {
         editor.selectionModel.addSelectionListener(codewhispererSelectionListener)
         Disposer.register(states) { editor.selectionModel.removeSelectionListener(codewhispererSelectionListener) }
 
+        val codewhispererDocumentListener: DocumentListener = object : DocumentListener {
```
Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt:442]: @andrewyuq, I'm wondering if we can put this logic inside https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorListener.kt#L13.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt [hunk 2]
```diff
@@ -395,6 +400,20 @@ class CodeWhispererPopupManager {
             ACTION_EDITOR_BACKSPACE,
             CodeWhispererPopupBackspaceHandler(actionManager.getActionHandler(ACTION_EDITOR_BACKSPACE), states)
         )
+
+        val actions = ComponentUtil.getClientProperty(states.requestContext.editor.component, ACTIONS_KEY) ?: return
+        actions.forEach {
+            val shortcutSet = it.shortcutSet
+            val shortcuts = shortcutSet.shortcuts
+            if (shortcuts.isEmpty() || shortcuts[0] !is KeyboardShortcut) return@forEach
+            if ((shortcuts[0] as KeyboardShortcut).firstKeyStroke.keyCode == 9) {
```
Comment (Will-ShaoHua) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt:409]: Andrew, a naive q, where I can look up the keyCode (9 here)

Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt:410]: it's tab key, I will use a different implementation to make it more clear.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt (hunk 1 above)
Comment (andrewyuq) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt:443]: They are different document listeners, that document listener should be attached as soon as editor is created, but for this one it should only be attached when CodeWhisperer's popup is about to show. (i.e. listener's scope is limited to `states` which are just when CodeWhisperer's section is active.

**Comment by andrewyuq** [comment]
I found that the fix in https://github.com/aws/aws-toolkit-jetbrains/pull/3227 will also fix this issue(they are actually the same issue) Since we now make CodeWhisperer's accept and navigate action to be the top priority. so I will remove the relevant fix in this PR and only keep the other fix and it's relevant tests.

**Review by Will-ShaoHua — APPROVED** [review]
LGTM

**Review by rli — APPROVED** [review]

**Pull request merged**